### PR TITLE
Update application form -models

### PR DESCRIPTION
**Pull request by klemmari1**
- Upon closer inspection we decided to rename a field and add better verbose names.
- Add ApplicationMixin with fields for accepting and rejecting an application.
- Add new model for apartments and HASO application apartment priorities. Also modify the serializers, factories and tests to match these changes.

Refs ASU-355 ASU-356

### CODE REVIEW

File: application_form/models.py [hunk 1]
```diff
@@ -9,14 +9,55 @@
...
+
+    objects = ApplicationQuerySet.as_manager()
+
+    class Meta:
+        abstract = True
+
+    def save(self, **kwargs):
+        if self.is_approved and self.is_rejected:
+            raise ValueError(
+                _("application cannot be accepted and rejected at the same time.")
```
Comment (ehaivala) [inline application_form/models.py:57]: Typo in word "application"

File: application_form/models.py [hunk 2]
```diff
@@ -9,14 +9,55 @@
...
+            )
+        if self.applicant_has_accepted_offer and not self.is_approved:
+            raise ValueError(
+                _("the offer cannot be accepted before the applicaiton is approved.")
```
Comment (ehaivala) [inline application_form/models.py:61]: Typo in word "application"

Comment (klemmari1) [reply application_form/models.py]: Updated :+1:

File: application_form/models.py (hunk 1 above)
Comment (ehaivala) [reply application_form/models.py:57]: Could you update this as well? 😄

Comment (klemmari1) [reply application_form/models.py:57]: Hmm, I cannot see the typo here. So in the word "application"?

Comment (ehaivala) [reply application_form/models.py:57]: Uhh, never mind. I think the word application just started to look incorrect even if it was written correctly. 👀

**Review by ehaivala — APPROVED** [review]

**Pull request merged**